app/src/test/java/org/simple/clinic/setup/SetupActivityEffectHandlerTest.kt

```diff
@@ -34,15 +38,16 @@ class SetupActivityEffectHandlerTest {
   }
 
   @Test
-  fun `the user details must be fetched when the fetch user details effect is received`() {
+  fun `the user onboarding status must be fetched when the fetch user details effect is received`() {
     // given
-    whenever(onboardingCompletePreference.get()).doReturn(true)
+    whenever(onboardingCompletePreference.get()) doReturn true
+    whenever(userDao.userImmediate()).doReturn<User?>(null)
 
     // when
     testCase.dispatch(FetchUserDetails)
 
     // then
-    testCase.assertOutgoingEvents(UserDetailsFetched(hasUserCompletedOnboarding = true))
+    testCase.assertOutgoingEvents(UserDetailsFetched(hasUserCompletedOnboarding = true, loggedInUser = None))
```

Comment:
Consider using factory functions to set these properties, gives user more context?

Reply:
Discussed offline: all three of the properties will always get set together and not individually. The tests have been updated to reflect this, so there was no need to create separate factory functions.